firstvoices/backend/views/category_views.py

```diff
@@ -44,11 +46,34 @@ def get_queryset(self):
     def get_list_queryset(self):
         site = self.get_validated_site()
         if site.count() > 0:
-            return (
-                Category.objects.filter(site__slug=site[0].slug)
-                .prefetch_related("children")
-                .exclude(parent__isnull=False)
-            )
+            list_queryset = Category.objects.filter(site__slug=site[0].slug)
+
+            # Check if type flags are present
+            contains_flags = self.request.GET.get("contains", "").split("|")
+
+            if all(
```

Comment:
I think this would be more straightforward and also more expandable as a list of filters instead of a series of if/else cases. Maybe something like:

for each supported type
      if the type is in the contains_flags, add the necessary Q filter to a list

type_filter = reduce(operator.or_, list_of_q_objects, Q())

list_queryset.filter(type_filter)
list_queryset.filter(parent__isnull=False)